### PR TITLE
Change PoET fork resolution

### PR DESCRIPTION
**Pull request by jjason**

### CODE REVIEW

**Comment by jjason** [comment]
To test, run the following:

`rm -rf ~/sawtooth/data`
`./bin/sawtooth config proposal create -k ~/sawtooth/keys/validator.wif sawtooth.consensus.algorithm=poet -o config.batch`
`./bin/poet genesis -k ~/sawtooth/keys/validator.wif -o poet_genesis.batch`
`./bin/sawtooth admin genesis config.batch poet_genesis.batch`
`./bin/validator -vv`

**Comment by jjason** [comment]
You can also run the PoET smoke test numerous times:

`cd integration/sawtooth_integration/docker/ && run_docker_test poet-smoke.yaml`

File: consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py [hunk 1]
```diff
@@ -58,5 +58,44 @@ def compare_forks(self, cur_fork_head, new_fork_head):
...
+            LOGGER.info(
+                'Signature of new fork head (%s...%s) > than current '
+                '(%s...%s)',
+                new_fork_head.header_signature[:8],
+                new_fork_head.header_signature[-8:],
+                cur_fork_head.header_signature[:8],
+                cur_fork_head.header_signature[-8:])
+            return True
+
+        LOGGER.info(
```
Comment (peterschwarz) [inline consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py]: Might be good to put this in an `else` block, for clarity.  Seems a bit odd all by itself.

File: consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py [hunk 2]
```diff
@@ -58,5 +58,44 @@ def compare_forks(self, cur_fork_head, new_fork_head):
             Boolean: True if the new chain should replace the current chain.
             False if the new chain should be discarded.
         """
-        LOGGER.debug("PoetForkResolver.compare_forks()")
-        return new_fork_head.block_num > cur_fork_head.block_num
+        if new_fork_head.block_num > cur_fork_head.block_num:
+            LOGGER.info(
```
Comment (peterschwarz) [inline consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py]: Should these be `debug`?

File: consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py (hunk 1 above)
Comment (jjason) [inline consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py]: Can do.  Seemed superfluous.

File: consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py (hunk 2 above)
Comment (jjason) [inline consensus/poet/core/sawtooth_poet/poet_consensus/poet_fork_resolver.py]: Yes.  However, while working on this I want those messages in the log to go with the chain/publisher messages associated with them.

**Comment by jjason** [comment]
Yeah.  I am hunting down why that is.  We have temporarily disable the PoET smoke test while I am debugging that so we can prevent create a logjam on PRs.

**Pull request merged**